src/page-multi-facility/ReSyncRefFacilityModal.tsx

```diff
@@ -0,0 +1,266 @@
+  border-bottom: 0;
+`;
+
+const FacilityName = styled.div`
+  flex: 1 1;
+`;
+
+const FacilitySelectContainer = styled.div`
+  max-width: 300px;
+  flex: 1 1;
+`;
+
+const Spacer = styled.span`
+  margin-right: 2em;
+`;
+
+const SyncDataTitle: React.FC<{
+  stateName: ModalProps["stateName"];
+  systemType: ModalProps["systemType"];
+}> = ({ stateName, systemType }) => (
+  <SyncDataTitleContainer>
+    <TitleText>Prepopulate Data</TitleText>
+    <TitleText>
+      We've found new facility data - select a corresponding facility to
+      autofill with case data
+    </TitleText>
+    {stateName && systemType && (
+      <TitleText>
+        State: {stateName}
+        <Spacer />
+        Type of System: {systemType}
+      </TitleText>
+    )}
+  </SyncDataTitleContainer>
+);
+
+const FacilitiesSelect: React.FC<{
+  value: Facility["id"] | undefined;
+  selections: { [key in ReferenceFacility["id"]]: Facility["id"] };
+  facilities: Facilities;
+  onChange: (value: Facility["id"] | undefined) => void;
+}> = ({ selections, value, facilities, onChange }) => {
+  const disabledOption = (facilityId: Facility["id"]) =>
+    Object.values(selections).includes(facilityId);
+  return (
+    <FacilitySelectContainer>
+      <InputSelect
+        onChange={(event) => {
+          console.log("onchange event", event.target.value);
+          const value = event.target.value;
+          onChange(value === "" ? undefined : value);
+        }}
+        value={value || ""}
+      >
+        <option value={""}>Select a facility</option>
+        {facilities.map((facility: Facility) => {
+          return (
+            <option
+              key={facility.id}
+              value={facility.id}
+              disabled={disabledOption(facility.id)}
+            >
+              {facility.name}
+            </option>
+          );
+        })}
+      </InputSelect>
+    </FacilitySelectContainer>
+  );
+};
+
+const filterSyncedFacilities = (
+  syncedRefFacilities: FacilityReferenceMapping,
+) => {
+  return (facility: Facility) => {
+    return !Object.keys(syncedRefFacilities).includes(facility.id);
+  };
+};
+
+interface ModalProps {
+  open: boolean;
+  stateName: ModelInputs["stateName"];
+  systemType: Facility["systemType"];
+  onClose: () => void;
+}
+
+const ReSyncRefFacilityModal: React.FC<ModalProps> = ({
+  open,
+  stateName,
+  systemType,
+  onClose,
+}) => {
+  const rejectionToast = useRejectionToast();
+  const [scenarioState, dispatchScenarioUpdate] = useScenario();
+  const {
+    state: { facilities: facilitiesState, referenceFacilities },
+  } = useFacilities();
+
+  const [selections, setSelections] = useState<{
+    [refFacilityId: string]: Facility["id"];
+  }>({});
+
+  const scenario = scenarioState.data;
+  const syncedRefFacilities = scenario?.[referenceFacilitiesProp] || {};
+
+  if (!open) return null;
+
+  const facilities = Object.values(facilitiesState).filter(
+    filterSyncedFacilities(syncedRefFacilities),
```

Comment:
`filterSyncedFacilities` isn't actually doing the filtering, right?  If not, maybe it would be better named something like `hasNoSyncedFacilities` since it's more of a check/conditional that ultimately gets used in a filter.